google-cloud-spanner-hibernate-dialect/src/main/java/com/google/cloud/spanner/hibernate/SpannerServiceContributor.java

```diff
@@ -45,19 +41,9 @@
   @Override
   public void contribute(StandardServiceRegistryBuilder serviceRegistryBuilder) {
     serviceRegistryBuilder
+        // The user agent JDBC connection property to identify the library.
         .applySetting("hibernate.connection.userAgent", HIBERNATE_API_CLIENT_LIB_TOKEN)
-        .addInitiator(
-            new StandardServiceInitiator<SchemaManagementTool>() {
-              @Override
-              public Class<SchemaManagementTool> getServiceInitiated() {
-                return SchemaManagementTool.class;
-              }
-
-              @Override
-              public SpannerSchemaManagementTool initiateService(
-                  Map configurationValues, ServiceRegistryImplementor registry) {
-                return SCHEMA_MANAGEMENT_TOOL;
-              }
-            });
+        // The custom Hibernate schema management tool for Spanner.
+        .applySetting("hibernate.schema_management_tool", SCHEMA_MANAGEMENT_TOOL);
```

Comment:
If one specifies it in the properties as well, which one takes precedence?

Reply:
The service contributor takes precedence :/